ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -523,13 +584,17 @@ BlobReadOptions getBlobReadInfo(StoreKey id, EnumSet<StoreGetOptions> getOptions
    * Gets {@link BlobReadOptions} for a deleted blob.
    * @param value the {@link IndexValue} of the delete index entry for the blob.
    * @param key the {@link StoreKey} for which {@code value} is the delete {@link IndexValue}
+   * @param indexSegments the map of index segment start {@link Offset} to {@link IndexSegment} instances
    * @return the {@link BlobReadOptions} that contains the information for the given {@code id}
    * @throws StoreException
    */
-  private BlobReadOptions getDeletedBlobReadOptions(IndexValue value, StoreKey key) throws StoreException {
+  private BlobReadOptions getDeletedBlobReadOptions(IndexValue value, StoreKey key,
+      ConcurrentSkipListMap<Offset, IndexSegment> indexSegments) throws StoreException {
     BlobReadOptions readOptions;
     try {
-      IndexValue putValue = findKey(key, new FileSpan(getStartOffset(), value.getOffset()), IndexEntryType.PUT);
+      IndexValue putValue =
+          findKey(key, new FileSpan(getStartOffset(indexSegments), value.getOffset()), IndexEntryType.PUT,
+              indexSegments);
       if (value.getOriginalMessageOffset() != -1) {
         // PUT record in the same log segment.
         String logSegmentName = value.getOffset().getName();
```

Comment:
What we concluded from the offline discussion was that this isn't necessary. We have no problems with correctness and the guarantees provided by this function are not changing.